src/gtclang/Frontend/GTClangASTVisitor.cpp

```diff
@@ -27,6 +27,25 @@ GTClangASTVisitor::GTClangASTVisitor(GTClangContext* context)
     : context_(context), globalVariableParser_(context),
       stencilParser_(context, globalVariableParser_) {}
 
+// We try to find all the interval-definitions
+bool GTClangASTVisitor::VisitDecl(clang::Decl* Decl) {
+  const auto& SM = context_->getSourceManager();
+  // Only parse declaration of the main-file
+  if(SM.getFileID(Decl->getLocation()) == SM.getMainFileID()) {
+    // We need it to be a var-decl
+    if(llvm::isa<clang::VarDecl>(Decl)) {
+      clang::VarDecl* vd = llvm::dyn_cast<clang::VarDecl>(Decl);
+      if(vd->getType().getBaseTypeIdentifier()) {
+          // we check for interval-defn's that are not the autogenerated loopvariables
+        if(vd->getType().getBaseTypeIdentifier()->getName() == "interval" && vd->getName()!="__k_loopvar__") {
```

Comment:
what is this k_loopvar?

Reply:
the prerprocessor replaces instances of 
```
vertical_region(k_begin, k_finish)
```
with 
```
for(auto __k_loopvar__ : {k_begin, k_finish}
```
and therefore it has the same type and would have been removed.